src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1430,6 +1425,476 @@ mfem::Geometry::Type MFEMSidreDataCollection::getElementTypeFromName(
   }
 }
 
+  #if defined(AXOM_USE_MPI) && defined(MFEM_USE_MPI)
+namespace detail
+{
+/**
+ * @brief Wrapper class to enable the reconstruction of a ParMesh from a
+ * Conduit blueprint
+ * 
+ * @note This is only needed because mfem::ParMesh doesn't provide a
+ * constructor that allows us to set all the required information.
+ */
+class SidreParMeshWrapper : public mfem::ParMesh
+{
+public:
+  /**
+   * @brief Constructs a new parallel mesh, intended to be assigned
+   * to an mfem::ParMesh*
+   * 
+   * @param [in] bp_grp The root group of the Conduit Blueprint structure
+   * @param [in] comm The MPI communicator to use with the new mesh
+   * @param [in] mesh_topo_name The name of the mesh topology in the Blueprint structure
+   * 
+   * The remaining parameters are used to construct the mfem::Mesh base subobject
+   */
+  SidreParMeshWrapper(Group* bp_grp,
+                      MPI_Comm comm,
+                      const std::string& mesh_topo_name,
+                      double* vertices,
+                      int num_vertices,
+                      int* element_indices,
+                      mfem::Geometry::Type element_type,
+                      int* element_attributes,
+                      int num_elements,
+                      int* boundary_indices,
+                      mfem::Geometry::Type boundary_type,
+                      int* boundary_attributes,
+                      int num_boundary_elements,
+                      int dimension,
+                      int space_dimension = -1)
+  {
+    MyComm = comm;
+    gtopo.SetComm(comm);
+    ConstructMeshSubObject(vertices,
+                           num_vertices,
+                           element_indices,
+                           element_type,
+                           element_attributes,
+                           num_elements,
+                           boundary_indices,
+                           boundary_type,
+                           boundary_attributes,
+                           num_boundary_elements,
+                           dimension,
+                           space_dimension);
+
+    SLIC_ERROR_IF(!bp_grp->hasGroup("adjsets"),
+                  "Cannot reconstruct a ParMesh without adjacency sets");
+
+    MPI_Comm_size(MyComm, &NRanks);
+    MPI_Comm_rank(MyComm, &MyRank);
+
+    // NOTE: The remaining logic is borrowed heavily from the mfem::ParMesh
+    // constructors, since it's setting up the internal data structures used
+    // to keep track of communication groups and shared elements
+
+    ReduceMeshGen();  // determine the global 'meshgen'
+
+    // FIXME: The word "group" is overloaded in this context
+    // Can we do any better with variable naming?
+    // A group can refer to a communication group or a Sidre group
+
+    auto mesh_adjset_groups =
```

Comment:
This is a good point. should we clarify it in the variable names when we can? sidre_group or whatever?

Reply:
Unfortunately we don't have a lot of options here because we're initializing the member variables defined in MFEM - there's a few places where I can tweak it though